analyses/sample-distribution-analysis/02-multilayer-pie.R

```diff
@@ -0,0 +1,91 @@
+# This script creates a treemap and a multilayer pie chart to represent the
+# broad histologies, short histologies, and molecular subtypes within the dataset.
+#
+# This script uses the packages sunburstR, d3r, and treemap to produce the
+# visualizations.
+#
+# Chante Bethell for CCDL 2019
+
+# Load/install packages. This will be removed once a Dockerfile is established.
+source("00-install-packages.R", print.eval = FALSE)
+
+# Load source script
+source("01-filter-across-types.R")
+
+# Set path to plots directory
+outputDir <- file.path("/plots")
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Create a colorblind-friendly color vector
+color <- colorblindr::palette_OkabeIto
+
+sun_df <- df2 %>%
+  dplyr::select(broad_histology, short_histology, disease_type_new) %>%
+  dplyr::filter(!is.na(disease_type_new)) %>%
+  dplyr::filter(!is.na(broad_histology)) %>%
+  dplyr::filter(!is.na(short_histology)) %>%
+  dplyr::group_by(broad_histology)
+
+sun_df_sub <-
+  sun_df[, c("broad_histology", "short_histology", "disease_type_new")]
+sun_df_sub$nodes <-
+  paste0(sun_df_sub[[1]], ",", sun_df_sub[[2]], ",", sun_df_sub[[3]])
+
+# Make node patterns
+names(sun_df_sub) <- c("level1", "level2", "level3", "nodes")
```

Comment:
I would probably make what is happening here more explicit:
```suggestion
colnames(sun_df_sub) <- c("level1", "level2", "level3", "nodes")
```